SlaveWAMPServer.spec.js

```diff
@@ -211,10 +209,305 @@ describe('SlaveWAMPServer', () => {
+			expect(slaveWampServer.interProcessRPC[validRequestB.socketId])
+				.to.have.property(validRequestB.procedure);
+			expect(slaveWampServer.interProcessRPC[validRequestB.socketId][validRequestB.procedure])
+				.to.have.property(validRequestB.signature).to.be.a('function');
+		});
+	});
+
+	describe('deleteCall', () => {
+		let validSocketId;
+		let validProcedure;
+		let validSignature;
+		let validRequest;
+
+		beforeEach(() => {
+			validSocketId = 'validSocketId';
+			validProcedure = 'validProcedure';
+			validSignature = 'validSignature';
+			validRequest = {
+				socketId: validSocketId,
+				procedure: validProcedure,
+				signature: validSignature,
+			};
+		});
+
+		it('should throw an error when invoked without arguments', () => {
+			expect(() => {
+				slaveWampServer.deleteCall();
+			}).to.throw('Cannot delete a call for wrong InterProcessRPCRequest request');
+			expect(slaveWampServer.interProcessRPC).to.be.empty();
+		});
+
+		it('should throw an error when invoked without socket id', () => {
+			delete validRequest.socketId;
+			expect(() => {
+				slaveWampServer.deleteCall(validRequest);
+			}).to.throw('Cannot delete a call for wrong InterProcessRPCRequest request');
+			expect(slaveWampServer.interProcessRPC).to.be.empty();
+		});
+
+		it('should throw an error when invoked without procedure', () => {
+			delete validRequest.procedure;
+			expect(() => {
+				slaveWampServer.deleteCall(validRequest);
+			}).to.throw('Cannot delete a call for wrong InterProcessRPCRequest request');
+			expect(slaveWampServer.interProcessRPC).to.be.empty();
+		});
+
+		it('should throw an error when invoked without signature', () => {
+			delete validRequest.signature;
+			expect(() => {
+				slaveWampServer.deleteCall(validRequest);
+			}).to.throw('Cannot delete a call for wrong InterProcessRPCRequest request');
+			expect(slaveWampServer.interProcessRPC).to.be.empty();
+		});
+
+		it('should throw an error when entry does not exist', () => {
+			expect(() => {
+				slaveWampServer.deleteCall(validRequest);
+			}).to.throw('There is no internal requests registered for socket: validSocketId, procedure: validProcedure with signature validSignature');
+		});
+
+		describe('when call exists', () => {
+			beforeEach(() => {
+				slaveWampServer.interProcessRPC = {
+					[validSocketId]: {
+						[validProcedure]: {
+							[validSignature]: () => {},
+						},
+					},
+				};
+			});
+
+			it('should delete existing signature call for procedure in interProcessRPC', () => {
+				slaveWampServer.deleteCall(validRequest);
+				expect(slaveWampServer.interProcessRPC[validSocketId][validProcedure]).to.be.empty();
+			});
+		});
+	});
+
+	describe('getCall', () => {
+		let validSocketId;
+		let validProcedure;
+		let validSignature;
+		let validRequest;
+
+		beforeEach(() => {
+			validSocketId = 'validSocketId';
+			validProcedure = 'validProcedure';
+			validSignature = 'validSignature';
```

Comment:
These same strings are reused in lots of different blocks - it would be neater to define them (as `const`) in an outer block.